instrumentation/kafka/kafka-clients/kafka-clients-common/library/src/main/java/io/opentelemetry/instrumentation/kafkaclients/OpenTelemetryKafkaMetrics.java

```diff
@@ -0,0 +1,150 @@
+ * //    config.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, ...);
+ * //    ...
+ * //    try (KafkaProducer<byte[], byte[]> producer = new KafkaProducer<>(config)) { ... }
+ * }</pre>
+ */
+public class OpenTelemetryKafkaMetrics implements MetricsReporter {
+
+  private static final Logger logger = Logger.getLogger(OpenTelemetryKafkaMetrics.class.getName());
+  @Nullable private static volatile Meter meter;
+
+  private static final List<RegisteredObservable> registeredObservables = new ArrayList<>();
+
+  /**
+   * Setup OpenTelemetry. This should be called as early in the application lifecycle as possible.
+   * Kafka metrics that are observed before this is called may not be bridged.
+   */
+  public static void setOpenTelemetry(OpenTelemetry openTelemetry) {
+    meter = openTelemetry.getMeter("io.opentelemetry.kafka-clients");
+  }
+
+  /**
+   * Reset for test by reseting the {@link #meter} to {@code null} and closing all registered
+   * instruments.
+   */
+  static void resetForTest() {
+    meter = null;
+    closeAllInstruments();
+  }
+
+  // Visible for test
+  static List<RegisteredObservable> getRegisteredObservables() {
+    return registeredObservables;
+  }
+
+  @Override
+  public void init(List<KafkaMetric> metrics) {
+    metrics.forEach(this::metricChange);
+  }
+
+  @Override
+  public synchronized void metricChange(KafkaMetric metric) {
+    KafkaMetricId kafkaMetricId = KafkaMetricId.create(metric);
+    Meter currentMeter = meter;
+    if (currentMeter == null) {
+      logger.log(Level.FINEST, "Metric changed but meter not set: {0}", kafkaMetricId);
+      return;
+    }
+
+    RegisteredObservable registeredObservable =
+        KafkaMetricRegistry.getRegisteredObservable(currentMeter, kafkaMetricId, metric);
+    if (registeredObservable == null) {
+      logger.log(Level.FINEST, "Metric changed but cannot map to instrument: {0}", kafkaMetricId);
+      return;
+    }
+
+    InstrumentDescriptor instrumentDescriptor = registeredObservable.getInstrumentDescriptor();
+    Attributes attributes = registeredObservable.getAttributes();
+    Set<AttributeKey<?>> attributeKeys = attributes.asMap().keySet();
+
+    for (Iterator<RegisteredObservable> it = registeredObservables.iterator(); it.hasNext(); ) {
+      RegisteredObservable curRegisteredObservable = it.next();
+      Set<AttributeKey<?>> curAttributeKeys =
+          curRegisteredObservable.getAttributes().asMap().keySet();
+      if (curRegisteredObservable.equals(registeredObservable)) {
+        logger.log(Level.FINEST, "Replacing instrument: {0}", curRegisteredObservable);
+        closeInstrument(curRegisteredObservable.getObservable());
+        it.remove();
+      } else if (curRegisteredObservable.getInstrumentDescriptor().equals(instrumentDescriptor)
+          && attributeKeys.size() > curAttributeKeys.size()
+          && attributeKeys.containsAll(curAttributeKeys)) {
```

Comment:
Here's how I detect whether a metric exists with less granular set of attribute keys, as explained in [this part of the readme](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/6138/files#r904288643).

In the process of adding support for this, I shuffled some stuff around for better organization, including moving the table print method to `OpenTelemetryKafkaMetricTest`. It turns out that all the information needed to print the table can be obtained without any additional surface area.